server/testutil/testenv/testenv.go

```diff
@@ -167,7 +167,10 @@ func GetTestEnv(t testing.TB) *real_environment.RealEnv {
 	}
 
 	healthChecker := healthcheck.NewHealthChecker("test")
-	t.Cleanup(healthChecker.Shutdown)
+	t.Cleanup(func() {
+		healthChecker.Shutdown()
+		healthChecker.WaitForGracefulShutdown()
```

Comment:
`WaitForGracefulShutdown` also starts the health check loop (IMO it should be named `RunHealthChecksUntilShutdown` or something like that). In https://github.com/buildbuddy-io/buildbuddy/pull/10154/files I proposed changing `Shutdown` to block until shutdown completes - wdyt about that approach?

Reply:
Talked offline; resolving.